CHORE: Fix nondescript pkg error and other issues

skarekrow opened the pull request:
Things addressed:
- Typos, whitespace and more line wraps (Yes I’m old school with 80.)
- Only show that we’re listening on websocket when a server is requested
- Fix Full Update not working
- Remove allText usages in update.go
- Iterate over both stdout and stderr for the main upgrade process
- Add utils module that currently allows us to easily copy files

Signed-off-by: Brandon Schneider <brandon@ixsystems.com>

Commit:
CHORE: Fix nondescript pkg error and other issues

Things addressed:
- Typos, whitespace and more line wraps (Yes I’m old school with 80.)
- Only show that we’re listening on websocket when a server is requested
- Fix Full Update not working
- Remove allText usages in update.go
- Iterate over both stdout and stderr for the main upgrade process
- Add utils module that currently allows us to easily copy files

Signed-off-by: Brandon Schneider <brandon@ixsystems.com>

Commit:
Remove debug

Signed-off-by: Brandon Schneider <brandon@ixsystems.com>

kmoore134 commented on update/update.go:
```diff
@@ -527,17 +541,40 @@ func checkbaseswitch() {
+			"Copying /etc/resolv.conf failed: %s\n", err,
+		)
+		logger.LogToFile(err_string)
+		ws.SendFatalMsg(err_string)
```
Do you want to also stop the update here? Sending the fatal msg will stop the client, but won't stop if running in WS only mode.

Commit:
Remove redundant ERROR

These already have that prepended from the call

Signed-off-by: Brandon Schneider <brandon@ixsystems.com>

Commit:
Update README with changes

Signed-off-by: Brandon Schneider <brandon@ixsystems.com>

Commit:
Address feedback and more

- We now return an error interface with the string of what we sent the WS
- DestroyMdDev for any error so as not to leave dangling mounts
- Correct error on failing to capture stdout or stderr
- Removed two instances of unused err
- Be more consistent with bootloader
- Remove SendBlMsg, replace it with an optional bool in SendInfoMsg
- Remove redundant JSONReply structs and specify it once in the module
- Wrap more lines

Signed-off-by: Brandon Schneider <brandon@ixsystems.com>

Commit:
Correctly let the consumer know about invalid JSON

Currently only the CLI prints.

Signed-off-by: Brandon Schneider <brandon@ixsystems.com>
